gateway/src/apicast/policy/token_introspection/token_introspection.lua

```diff
@@ -47,6 +56,13 @@ function _M.new(config)
   return self
 end
 
+function _M:rewrite(context)
+  if self.use_oidc_issuer_endpoint then
+    local components = resty_url.parse(context.service.oidc.issuer_endpoint)
+    set_client(self, components.user, components.password)
```

Comment:
It might be easier to pass the headers into the individual http call instead of initializing new client every time. Then we won't mutate `self` and initialize new http client on every request.

You can pass headers when making the request: https://github.com/3scale/apicast/blob/154dc5a4087c97ef9c46209ada6fc8d7d944fea6/spec/resty/http_ng_spec.lua#L78

Reply:
Changed.